pkg/build/builder/util.go

```diff
@@ -46,26 +55,47 @@ func readNetClsCGroup(reader io.Reader) string {
 	names := []string{"net_cls", "cpu"}
 	for _, group := range names {
 		if value, ok := cgroups[group]; ok {
-			return value
+			return value, containerType
 		}
 	}
 
-	return ""
+	return "", containerType
 }
 
-// getDockerNetworkMode determines whether the builder is running as a container
+// getContainerNetworkConfig determines whether the builder is running as a container
 // by examining /proc/self/cgroup. This context is then passed to source-to-image.
-func getDockerNetworkMode() s2iapi.DockerNetworkMode {
+// It returns a suitable argument for NetworkMode.  If the container platform is
+// CRI-O, it also returns a path for /etc/resolv.conf, suitable for bindmounting.
+func getContainerNetworkConfig() (string, string, error) {
 	file, err := os.Open("/proc/self/cgroup")
 	if err != nil {
-		return ""
+		return "", "", err
 	}
 	defer file.Close()
 
-	if id := readNetClsCGroup(file); id != "" {
-		return s2iapi.NewDockerNetworkModeContainer(id)
+	if id, containerType := readNetClsCGroup(file); id != "" {
+		glog.V(5).Infof("container type=%s", containerType)
+		if containerType != "crio-" {
+			return s2iapi.DockerNetworkModeContainerPrefix + id, "", nil
+		}
+
+		crioClient, err := crioclient.New(crio.CrioSocket)
+		if err != nil {
+			return "", "", err
+		}
+		info, err := crioClient.ContainerInfo(id)
+		if err != nil {
+			return "", "", err
+		}
+		pid := strconv.Itoa(info.Pid)
+		resolvConfHostPath := info.CrioAnnotations["io.kubernetes.cri-o.ResolvPath"]
```

Comment:
We can use this https://github.com/kubernetes-incubator/cri-o/blob/master/pkg/annotations/annotations.go#L53 from cri-o pkg/annotations.

Reply:
updated.